modules/templates/templates/templates.test.tsx

```diff
@@ -0,0 +1,732 @@
+import "@testing-library/jest-dom";
+import { render, screen, waitFor, within } from "@testing-library/react";
+import Modal from "react-modal";
+import userEvent from "@testing-library/user-event";
+import { CoreComponent, GlobalCore } from "@/core/module/module.types";
+import "./index";
+import "@/services/auth/auth-mock.service";
+import "@/services/templates/templates-mock.service";
+import { SupabaseTemplateService } from "@/core/module/services.types";
+
+import React, { act } from "react";
+import { RouterMock } from "@/jest-setup";
+import { getComponent, setRouteQuery } from "@/resources/tests/test.utils";
+import { Seed } from "@/resources/tests/seed";
+import {
+  GenesisTemplate,
+  GenesisTemplateField,
+  TYPE_OPTIONS,
+} from "@/core/module/core.types";
+
+describe("===== TEMPLATES =====", () => {
+  let templatesService: SupabaseTemplateService;
+
+  beforeAll(() => {
+    templatesService = GlobalCore.manager.getComponent(
+      "service",
+      "templates",
+    ) as SupabaseTemplateService;
+  });
+
+  describe("Templates Page", () => {
+    let Templates: CoreComponent;
+    beforeAll(() => {
+      Templates = getComponent("app", "templates");
+    });
+
+    beforeEach(() => {});
+    afterEach(() => {});
+
+    it("Templates is Mounted", async () => {
+      await act(() => render(<Templates />));
+
+      expect(screen.getByTestId("templates.title")).toBeInTheDocument();
+      expect(screen.getByTestId("templates.new-template")).toBeInTheDocument();
+      expect(screen.queryByText("templates.table")).not.toBeInTheDocument();
+      screen.debug();
+    });
+
+    it("Checks table is populated", async () => {
+      render(<Templates />);
+
+      // Wait for the table to be fully rendered (at least one row is filled)
+      await waitFor(() => {
+        expect(screen.getByTestId("table-row-0"));
+      });
+    });
+
+    it("Checks new entry is created in edit mode when add template", async () => {
+      render(<Templates />);
+
+      await waitFor(() => {
+        expect(screen.getByTestId("table-row-0"));
+      });
+
+      const templateEdit = screen.getByTestId("templates.new-template");
```

Comment:
The good practicse is to use testId when we have no better way.

I suggest lets try to get the button by text (byRole("button" , {"name" : "text-here"})